app/services/submit_application_choice.rb

```diff
@@ -14,7 +14,10 @@ def edit_by_time
     if HostingEnvironment.sandbox_mode?
       Time.zone.now
     else
-      TimeLimitConfig.edit_by.days.after(@application_choice.application_form.submitted_at).end_of_day
+      days = TimeLimitConfig.edit_by
+      date = days.type == :calendar ? days.count.days : days.count.business_days
+
+      date.after(@application_choice.application_form.submitted_at).end_of_day
```

Comment:
Feels like this could be encapsulated inside the `Days` class? Finding a suitable method name though... `to_days`?

```ruby
# In app/lib/time_limit_config.rb
def to_days
  type == :calendar ? count.days : count.business_days
end
```

Then this could be:

```ruby
TimeLimitConfig.edit_by.to_days.after(@application_choice.application_form.submitted_at).end_of_day
```

Reply:
Agreed! 🚀

Reply:
Resolved in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1765/commits/bf49496e8900d59b89d9d96cf71b77b6304bdc3f and https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1765/commits/b2e80a73bdfa6f004c57d4c5500fcaaa36a58ee4.